core/src/main/java/org/elasticsearch/index/mapper/internal/IdFieldMapper.java

```diff
@@ -306,12 +310,13 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
             return builder;
         }
         boolean includeDefaults = params.paramAsBoolean("include_defaults", false);
+        boolean hasCustomFieldDataSettings = customFieldDataSettings != null && customFieldDataSettings.equals(Settings.EMPTY) == false;
```

Comment:
I guess we can abstract this into a method called `hasCustomFieldDataSettings()` that way we don't have to repeat it like 7 times?